scalameta/contrib/src/main/scala/scala/meta/contrib/implicits/Statements.scala

```diff
@@ -0,0 +1,105 @@
+package scala.meta.contrib.implicits
+
+import scala.collection.immutable.Seq
+import scala.language.higherKinds
+import scala.language.implicitConversions
+import scala.meta._
+import scala.meta.contrib.Equal
+
+/**
+  * Typeclass to deal with scalameta trees which have an expected set of statements.
+  *
+  * For trees that have multiple stat lists (such as Defn.Class) the main stats are used.
+  *
+  * This allows you to do things like add a statement to the
+  * body of a class without digging through the low level meta api
+  *
+  * TODO: Consider using a meta variant of simulacrum to
+  * generate the boilerplate once one is avaliable
+  */
+trait Statements[A] {
+  def getStats(a: A): Seq[Stat]
+  def replaceStats(a: A, newStats: Seq[Stat]): A
+}
+
+object Statements {
+
+  trait Ops[A] {
+    val self: A
+    val typeClassInstance: Statements[A]
+
+    def getStats: Seq[Stat] =
```

Comment:
cant, it breaks when the tree itself implements a field stats. Thats the only reason i switched it